core/src/test/java/io/questdb/griffin/AlterTableAttachPartitionTest.java

```diff
@@ -1091,6 +1091,372 @@ public void testDetachAttachDifferentPartitionTableReaderReload() throws Excepti
+                lastModified.put(path.toString(), ff.getLastModified(path));
+            });
+
+            // execute an update directly on the cold storage partition
+            executeOperation(
+                    "UPDATE " + tableName + " SET l = 13 WHERE ts = '" + partitionName + "T00:00:17.279900Z'",
+                    CompiledQuery.UPDATE, CompiledQuery::getUpdateOperation
+            );
+            assertSql("SELECT min(ts), max(ts), count() FROM " + tableName,
+                    "min\tmax\tcount\n" +
+                            "2022-10-17T00:00:17.279900Z\t2022-10-18T23:59:59.000000Z\t10000\n");
+            assertSql("SELECT * FROM " + tableName + " WHERE ts in '" + partitionName + "' LIMIT 5",
+                    "l\ti\tts\n" +
+                            "13\t1\t2022-10-17T00:00:17.279900Z\n" +
+                            "2\t2\t2022-10-17T00:00:34.559800Z\n" +
+                            "3\t3\t2022-10-17T00:00:51.839700Z\n" +
+                            "4\t4\t2022-10-17T00:01:09.119600Z\n" +
+                            "5\t5\t2022-10-17T00:01:26.399500Z\n"
+            );
+
+            // verify that no new partition folder has been created in the table data space (hot)
+            // but rather the files in cold storage have been modified
+            path.of(configuration.getRoot())
+                    .concat(tableName)
+                    .concat(partitionName);
+            TableUtils.txnPartitionConditionally(path, txn - 1);
+            Assert.assertTrue(Files.exists(path.$()));
+            Assert.assertTrue(Os.type == Os.WINDOWS || Files.isSoftLink(path));
+            // in windows, detecting a soft link is tricky, and unnecessary. Removing
+            // a soft link does not remove the target's content, so we do not need to
+            // call unlink, thus we do not need isSoftLink. It has been implemented however
+            // and it does not seem to work.
+            path.of(other);
+            ff.walk(other, (file, type) -> {
+                // TODO: Update does not follow the usual path, like insert. To be able to
+                //  prevent modifications on cold storage we must be able to detect whether
+                //  the column files being versioned belong in a folder that is soft linked,
+                //  and then move first the data to hot storage and take it from there.
+                //  OR accept that UPDATE does modify cold storage, OR simply provide the
+                //  mechanism to flag a partition as RO/RW and fail updates on RO partitions.
+                //  The later will be achieved in a later PR.
+                path.trimTo(len).concat(file).$();
+                Long lm = lastModified.get(path.toString());
+                Assert.assertTrue(lm == null || lm.longValue() == ff.getLastModified(path));
+            });
+        });
+    }
+
+    @Test
+    public void testAlterTableAttachPartitionFromSoftLinkThenDropIt() throws Exception {
```

Comment:
Please add the test where sym link partition drop / detach is completed async via `O3PartitionPurgeJob`, see examples in `O3PartitionPurgeTest`

Reply:
yeah, indeed